crates/vrm-spec/src/vrmc_vrm_animation_1_0.rs

```diff
@@ -0,0 +1,140 @@
+//! Data structures for the [`VRMC_vrm_animation`](https://github.com/vrm-c/vrm-specification/blob/master/specification/VRMC_vrm_animation-1.0/README.md) 1.0 glTF Extension.
+
+#[cfg(feature = "rustc_hash")]
+use rustc_hash::FxHashMap as HashMap;
+use serde::{Deserialize, Serialize};
+#[cfg(not(feature = "rustc_hash"))]
+use std::collections::HashMap;
+
+/// VRMC_vrm_animation extension name
+pub const VRMC_VRM_ANIMATION: &str = "VRMC_vrm_animation";
+
+/// glTF extension that defines humanoid animations.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct VrmcVrmAnimationSchema {
+    /// Specification version of VRMC_vrm_animation
+    pub spec_version: String,
+
+    /// An object which describes about humanoid bones.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub humanoid: Option<VrmcVrmAnimationHumanoid>,
+
+    /// An object which maps expressions to nodes.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub expressions: Option<VrmcVrmAnimationExpressions>,
+
+    /// An object which maps a eye gaze point to a node.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub look_at: Option<VrmcVrmAnimationLookAt>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub extensions: Option<HashMap<String, HashMap<String, Option<serde_json::Value>>>>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub extras: Option<serde_json::Value>,
+}
+
+/// An object which describes about humanoid bones.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct VrmcVrmAnimationHumanoid {
+    /// An object which maps humanoid bones to nodes.
+    pub human_bones: VrmcVrmAnimationHumanBones,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub extensions: Option<HashMap<String, HashMap<String, Option<serde_json::Value>>>>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub extras: Option<serde_json::Value>,
+}
+
+/// Represents a set of humanBones of a humanoid.
+// FIXME: all bones are optional by now
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct VrmcVrmAnimationHumanBones(
+    pub HashMap<crate::vrmc_vrm_1_0::HumanBoneName, Option<VrmcVrmAnimationHumanBone>>,
+);
+
+/// Represents a single bone of a Humanoid.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct VrmcVrmAnimationHumanBone {
+    /// Represents a single glTF node tied to this humanBone.
+    #[cfg(feature = "gltf_index")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub node: Option<gltf::json::Index<gltf::json::Node>>,
```

Comment:
If unity puts -1 randomly in this field we need `deserialize_option_map_index`

Reply:
Is there a sample that has -1 in this field?

Reply:
I haven't encountered one yet but not sure.

Reply:
The existing codebase doesn't handle -1 for VRM 1.0, I think it's fine to leave it as is.